specification/under_development/profiles/Adaptive_Profile.md

```diff
@@ -0,0 +1,963 @@
+| __quantum__rt__initialize           | `void(i8*)`          | Initializes the execution environment. Sets all qubits to a zero-state if they are not dynamically managed.                                                                                                                                                 |
+| __quantum__rt__tuple_record_output  | `void(i64,i8*)`      | Inserts a marker in the generated output that indicates the start of a tuple and how many tuple elements it has. The second parameter defines a string label for the tuple. Depending on the output schema, the label is included in the output or omitted.  |
+| __quantum__rt__array_record_output  | `void(i64,i8*)`      | Inserts a marker in the generated output that indicates the start of an array and how many array elements it has. The second parameter defines a string label for the array. Depending on the output schema, the label is included in the output or omitted. |
+| __quantum__rt__result_record_output | `void(%Result*,i8*)` | Adds a measurement result to the generated output. The second parameter defines a string label for the result value. Depending on the output schema, the label is included in the output or omitted.                                                         |
+| __quantum__rt__bool_record_output   | `void(i1,i8*)`       | Adds a boolean value to the generated output. The second parameter defines a string label for the result value. Depending on the output schema, the label is included in the output or omitted.                                                              |
+
+The following output recording functions can appear if you opt into supporting
+real-time integer calculations on an integer or fixed-point `iN` type (the profile compliant program sets the following module flag `!{i32 1, !"classical_ints", i32 N}` or  `!{i32 1, !"classical_fixed_points", i32 N}`).
+
+| Function                         | Signature       | Description                                                                                                                                                                                              |
+| :------------------------------- | :-------------- | :------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| __quantum__rt__iN_record_output | `void(i64,i8*)` | Adds an integer result to the generated output. The second parameter defines a string label for the result value. Depending on the output schema, the label is included in the output or omitted.        |
+
+The following output recording functions can appear if you opt into supporting
+real-time floating point computations on an `fN` type (the profile compliant program sets the following module flag `!{i32 1, !"classical_ints", f32 N}`).
```

Comment:
This creates an unknown arbitrary set of recording calls to link against and provide. I think adding an i8 size param would simplify the runtime and tooling.